src/bin/cli/solana/deploy.rs

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::cli::check_target_match;
+use std::{process::exit, str::FromStr, time::Duration};
+
+use solana_cli::{
+    cli::{
+        process_command, CliCommand, CliCommandInfo, CliConfig, DEFAULT_CONFIRM_TX_TIMEOUT_SECONDS,
+        DEFAULT_RPC_TIMEOUT_SECONDS,
+    },
+    program::ProgramCliCommand,
+};
+use solana_cli_config::{Config, CONFIG_FILE};
+use solana_cli_output::OutputFormat;
+use solana_rpc_client_api::config::RpcSendTransactionConfig;
+use solana_sdk::{commitment_config::CommitmentConfig, signer::keypair::read_keypair_file};
+
+#[derive(Clone, Debug, clap::Args)]
+#[clap(name = "deploy", about = "Deploy a program to Solana")]
+pub struct SolanaDeploy {
+    #[clap(help = "Specifies the path to the program file to deploy (.so)")]
+    program_location: String,
+    #[clap(long, help = "Specifies whether to export the output in JSON format")]
+    output_json: bool,
+    #[clap(
+        short('v'),
+        long,
+        conflicts_with = "output_json",
+        help = "Specifies whether to display verbose program deployment information"
+    )]
+    verbose: bool,
+}
+
+impl SolanaDeploy {
+    /// Handle the deployment of a Solana program
+    ///
+    /// This function is responsible for managing the deployment process,
+    /// including checking the current directory, parsing command-line arguments,
+    /// configuring settings, and executing the deployment command. It also handles
+    /// loading the necessary configuration and signers, defining output formats,
+    /// and processing the deployment command using the provided configuration.
+    pub fn handle(&self) {
+        // Make sure the command is run in the correct directory
+        // Fails if the command is run in a Solang Polkadot project directory
+        if !check_target_match("solana", None).unwrap() {
```

Comment:
You removed all `unwrap()`s from `check_target_match`, instead returning an error. Which is desirable but rendered entirely pointless by this `unwrap()` here.